core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2582,7 +2586,7 @@ private[spark] class CallerContext(
         val callerContext = Class.forName("org.apache.hadoop.ipc.CallerContext")
         val builder = Class.forName("org.apache.hadoop.ipc.CallerContext$Builder")
         // scalastyle:on classforname
-        val builderInst = builder.getConstructor(classOf[String]).newInstance(context)
+        val builderInst = builder.getConstructor(classOf[String]).newInstance(prepareContext(context))
```

Comment:
Should prepareContext be done once as part of initialization of context or should it need to be done for each invocation of setCurrentContext ?